nmostesting/NMOSTesting.py

```diff
@@ -592,60 +598,78 @@ def parse_arguments():
     elif args.describe_suites:
         for test_suite in sorted(TEST_DEFINITIONS):
-            print(test_suite + ": " + TEST_DEFINITIONS[test_suite]["name"])
-        sys.exit(ExitCodes.OK)
-
-    if "suite" in vars(args):
-        if args.suite not in TEST_DEFINITIONS:
-            print(" * ERROR: The requested test suite '{}' does not exist".format(args.suite))
-            sys.exit(ExitCodes.ERROR)
+            msg += test_suite + ": " + TEST_DEFINITIONS[test_suite]["name"] + '\n'
+    elif "suite" in vars(args):
         if args.list_tests:
             tests = enumerate_tests(TEST_DEFINITIONS[args.suite]["class"])
             for test_name in tests:
-                print(test_name)
-            sys.exit(ExitCodes.OK)
-        if args.describe_tests:
+                msg += test_name + '\n'
+        elif args.describe_tests:
             tests = enumerate_tests(TEST_DEFINITIONS[args.suite]["class"], describe=True)
             for test_description in tests:
-                print(test_description)
-            sys.exit(ExitCodes.OK)
-        if args.selection and args.selection not in enumerate_tests(TEST_DEFINITIONS[args.suite]["class"]):
-            print(" * ERROR: Test with name '{}' does not exist in test definition '{}'"
-                  .format(args.selection, args.suite))
-            sys.exit(ExitCodes.ERROR)
-        if len(args.host) != len(args.port) or len(args.host) != len(args.version):
-            print(" * ERROR: Hostnames/IPs, ports and versions must contain the same number of elements")
-            sys.exit(ExitCodes.ERROR)
-        if len(args.host) != len(TEST_DEFINITIONS[args.suite]["specs"]):
-            print(" * ERROR: This test definition expects {} Hostnames/IP(s), port(s) and version(s)"
-                  .format(len(TEST_DEFINITIONS[args.suite]["specs"])))
-            sys.exit(ExitCodes.ERROR)
-        if args.output and not args.output.endswith("xml") and not args.output.endswith("json"):
-            print(" * ERROR: Output file must end with '.xml' or '.json'")
-            sys.exit(ExitCodes.ERROR)
+                msg += test_description + '\n'
+        elif args.suite not in TEST_DEFINITIONS:
+            msg = " * ERROR: The requested test suite '{}' does not exist".format(args.suite)
+            return_type = ExitCodes.ERROR
+        elif getattr(args, "selection", "all") not in enumerate_tests(TEST_DEFINITIONS[args.suite]["class"]):
+            msg = " * ERROR: Test with name '{}' does not exist in test definition '{}'".format(
+                args.selection, args.suite)
+            return_type = ExitCodes.ERROR
+        elif not args.host or not args.port or not args.version:
+            msg = " * ERROR: No Host(s) or Port(s) or Version(s) specified"
+            return_type = ExitCodes.ERROR
+        elif len(args.host) != len(args.port) or len(args.host) != len(args.version):
+            msg = " * ERROR: Hostnames/IPs, ports and versions must contain the same number of elements"

```

Comment:
Should we use either "host(s)" or "hostname(s)/IP address(es)" consistently in the messages?